File: banchus-packet/src/main/java/pe/nanamochi/banchus/packets/client/MatchJoinPacket.java
package pe.nanamochi.banchus.packets.client;

import java.io.IOException;
import java.io.InputStream;
import lombok.Data;
import lombok.NoArgsConstructor;
import pe.nanamochi.banchus.core.ClientPacket;
import pe.nanamochi.banchus.core.Packets;
import pe.nanamochi.banchus.io.IDataReader;

@Data
@NoArgsConstructor
public class MatchJoinPacket implements ClientPacket {
  private int matchId;
  private String matchPassword;

  @Override
  public Packets getPacketType() {
    return Packets.OSU_MATCH_JOIN;
  }

  @Override
  public void read(IDataReader reader, InputStream stream) throws IOException {
    this.matchId = reader.readInt32(stream);
    this.matchPassword = reader.readString(stream);
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/packets/client/MatchJoinPacket.kt
package pe.nanamochi.banchus.packets.client

import java.io.InputStream
import pe.nanamochi.banchus.core.BanchoPacket
import pe.nanamochi.banchus.core.PacketType
import pe.nanamochi.banchus.io.DataReader

class MatchJoinPacket : BanchoPacket(), BanchoPacket.Client {
    override val type = PacketType.OSU_MATCH_JOIN

    var matchId: Int = 0
    var matchPassword: String? = null

    override fun read(reader: DataReader, ins: InputStream) {
        this.matchId = reader.readInt32(ins)
        this.matchPassword = reader.readString(ins)
    }
}
